### PR TITLE
fix(Core/Spells): Totems should be immuned to others' area auras.

### LINKED ISSUE #14552: Shaman Totems should not receive Raid/Party Wide Auras
https://github.com/chromiecraft/chromiecraft/issues/4116
### What client do you play on?

enUS

### Faction

Alliance

### Content Phase:

60

### Current Behaviour

Shaman Totems receive Raid/Party Wide Auras
image

This causes combat log entries and apply/losing the aura and is very annoying for people with a scrolling combat text

### Expected Blizzlike Behaviour

Shaman Totems don't receive Raid/Party Wide Auras
image
Totem of Wrath, Mana Spring, Stoneskin and Wrath of Air are up
Stoneskin only has the Stoneskin aura

### Source

https://youtu.be/uJN4CiO7ewg?t=140

### Steps to reproduce the problem

Be Shaman
Place 2 Totems that gives auras
Target one of the totems

### Extra Notes

Non-combat/vanity pets also should not receive auras, but I cannot find proof for that

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/23f432c6b502c55c2e32749cd2b607d3eb752366

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-dmf-switch](https://github.com/azerothcore/mod-dmf-switch)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna](https://github.com/azerothcore/mod-eluna)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [mod-instanced-worldbosses](https://github.com/nyeriah/mod-instanced-worldbosses)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by UltraNix**
Fixes #14552

## Issues Addressed:

- Closes #14552
- Closes https://github.com/chromiecraft/chromiecraft/issues/4116

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

Be Shaman
Place 2 Totems that gives auras
Target one of the totems

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by IntelligentQuantum — APPROVED** [review]

**Comment by balleny** [comment]
the change must not prevent stoneskin totems from applying their protective buff on other totems
https://wowgaming.altervista.org/aowow/?search=stoneclaw+totem#abilities

File: src/server/game/Entities/Totem/Totem.cpp [hunk 1]
```diff
@@ -181,6 +181,12 @@ bool Totem::IsImmunedToSpellEffect(SpellInfo const* spellInfo, uint32 index) con
             spellInfo->Effects[index].TargetA.GetCheckType() != TARGET_CHECK_ENTRY && spellInfo->Id != 55277 && spellInfo->Id != 6277)
         return true;
 
+    // Immuned to other area raid auras
+    if (spellInfo->IsPositive() && spellInfo->Effects[index].IsAreaAuraEffect() && caster != this && IsFriendlyTo(caster))
```
Comment (robens) [inline src/server/game/Entities/Totem/Totem.cpp:185]: caster can be null for auras

**Review by Kitzunu — APPROVED** [review]

**Comment by Kitzunu** [comment]
Stoneskin aura isnt an area aura I think so it should be fine

**Comment by Kitzunu** [comment]
build failing

**Comment by Kitzunu** [comment]
abandoned

**Pull request closed without merging**